applications/finetune-quora-embeddings/benchmark.py

```diff
@@ -0,0 +1,354 @@
+import itertools
+from typing import List
+from modal import Image, Stub, Volume, Secret, gpu
+import os
+from sklearn.metrics import accuracy_score, precision_score, recall_score, roc_auc_score
+from evals import threshold_decorator
+from datasets import Dataset
+
+# Model Configuration
+MODELS = [
+    "BAAI/bge-base-en-v1.5",
+    "llmrails/ember-v1",
+    "thenlper/gte-large",
+    "infgrad/stella-base-en-v2",
+    "sentence-transformers/gtr-t5-large",
+]
+GPU_CONFIG = gpu.A100()
+
+# Eval COnfiguration
+METRICS = {
+    "accuracy": accuracy_score,  # This is the number of correct predictions by the model ( TP + TN )/ (# of samples)
+    "precision": precision_score,  # This measures the number of positive class predicitons (TP) / (TP + FP)
+    "recall": recall_score,  # This measure the number of negative class predictions (TP) / ( TP + FN )
+}
+THRESHOLDS = [0.3, 0.5, 0.7]
+EVALS = dict()
+for threshold, metric_name in itertools.product(THRESHOLDS, METRICS.keys()):
+    EVALS[f"{metric_name} ({threshold})"] = threshold_decorator(threshold)(
+        METRICS[metric_name]
+    )
+EVALS["AUC"] = roc_auc_score
+
+# Dataset Configuration
+DATASET_NAME = "567-labs/cleaned-quora-dataset-train-test-split"
+DATASET_DIR = "/data"
+DATASET_VOLUME = Volume.persisted("datasets")
+
+# Test Configuration
+TEST_PERCENTAGE = 0.1
+MAXIMUM_ELEMENTS_TO_TEST = 300
+COHERE_MODEL = "embed-english-v3.0"
+
+stub = Stub("cohere-embeddings")
+
+
+image = Image.debian_slim().pip_install(
+    "cohere", "datasets", "sentence-transformers", "scikit-learn", "tabulate", "openai"
```

Comment:
There's no download models function. 
Every time I run this I have to redownload every single model which takes time.